oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java

```diff
@@ -227,15 +239,27 @@ public static String getTokenServerEncodedUrl() {
     return getTokenServerEncodedUrl(DefaultCredentialsProvider.DEFAULT);
   }
 
+  @Override
+  public boolean createScopedRequired() {
+    return scopesRequired;
```

Comment:
I am not exactly sure if this is correct. This means `scopeRequired` is default to true but it was default to `true` before this change. This logic comes from `AppEngineCredentials`.